File: app/src/main/java/com/example/pam_project/features/tasks/create/CreateTaskActivity.java
package com.example.pam_project.features.tasks.create;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.CheckBox;
import android.widget.EditText;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import com.example.pam_project.R;
import com.example.pam_project.di.ApplicationContainer;
import com.example.pam_project.di.ApplicationContainerLocator;
import com.example.pam_project.repositories.tasks.TaskRepository;
import com.example.pam_project.utils.FormValidator;
import com.example.pam_project.utils.schedulers.SchedulerProvider;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class CreateTaskActivity extends AppCompatActivity implements CreateTaskView {
    private final static String LIST_ID_PARAMETER = "id";
    private CreateTaskPresenter presenter;
    private Long listId;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_create_task);
        createPresenter();

        listId = getIntent().getLongExtra(LIST_ID_PARAMETER, -1);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        Objects.requireNonNull(getSupportActionBar()).setTitle(R.string.activity_title_create_task);
    }

    private void createPresenter() {
        final ApplicationContainer container = ApplicationContainerLocator.locateComponent(this);
        final SchedulerProvider schedulerProvider = container.getSchedulerProvider();
        final TaskRepository taskRepository = container.getTasksRepository();
        presenter = new CreateTaskPresenter(schedulerProvider, taskRepository, this);
    }

    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.check_action_bar, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        long itemId = item.getItemId();
        final EditText taskNameInput = findViewById(R.id.create_task_title_input);
        final EditText taskDescriptionInput = findViewById(R.id.create_task_description_input);
        final CheckBox checkboxUrgencyInput = findViewById(R.id.create_task_priority_checkbox);

        if (itemId == R.id.check_add_button) {
            String taskName = taskNameInput.getText().toString();
            String taskDescription = taskDescriptionInput.getText().toString();
            boolean taskUrgency = checkboxUrgencyInput.isChecked();
            boolean validForm = FormValidator.validate(getApplicationContext(), createInputMap(taskName, taskNameInput));

            if (validForm) {
                presenter.insertTask(taskName, taskDescription, taskUrgency, listId);
                finish();
            }
        } else if (itemId == android.R.id.home) {
            onBackPressed();
        }

        return super.onOptionsItemSelected(item);
    }

    private Map<EditText, String> createInputMap(String taskName, EditText taskNameInput) {
        Map<EditText, String> map = new HashMap<>();
        map.put(taskNameInput, taskName);
        return map;
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }

    @Override
    public void onStop() {
        super.onStop();
        presenter.onViewDetached();
    }
}

File: app/src/main/java/com/example/pam_project/features/tasks/create/CreateTaskActivity.kt
package com.example.pam_project.features.tasks.create

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.CheckBox
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.pam_project.R
import com.example.pam_project.di.ApplicationContainerLocator
import com.example.pam_project.utils.validators.FormValidator
import java.util.*

class CreateTaskActivity : AppCompatActivity(), CreateTaskView {
    private lateinit var presenter: CreateTaskPresenter
    private var listId: Long? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_create_task)
        createPresenter()
        listId = intent.getLongExtra(LIST_ID_PARAMETER, -1)
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title=R.string.activity_title_create_task.toString()
    }

    private fun createPresenter() {
        val container = ApplicationContainerLocator.locateComponent(this)
        val schedulerProvider = container.schedulerProvider
        val taskRepository = container.tasksRepository
        presenter = CreateTaskPresenter(schedulerProvider, taskRepository, this)
    }

    public override fun onStart() {
        super.onStart()
    }

    override fun onTaskInsertedError() {
        Toast.makeText(applicationContext, getString(R.string.error_task_create), Toast.LENGTH_LONG).show()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.check_action_bar, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val itemId = item.itemId.toLong()
        val taskNameInput = findViewById<EditText>(R.id.create_task_title_input)
        val taskDescriptionInput = findViewById<EditText>(R.id.create_task_description_input)
        val checkboxUrgencyInput = findViewById<CheckBox>(R.id.create_task_priority_checkbox)
        if (itemId == R.id.check_add_button.toLong()) {
            val taskName = taskNameInput.text.toString()
            val taskDescription = taskDescriptionInput.text.toString()
            val taskUrgency = checkboxUrgencyInput.isChecked
            val validForm = FormValidator.validate(applicationContext, createInputMap(taskName, taskNameInput))
            if (validForm) {
                presenter.insertTask(taskName, taskDescription, taskUrgency, listId!!)
                finish()
            }
        } else if (itemId == android.R.id.home.toLong()) {
            onBackPressed()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun createInputMap(taskName: String, taskNameInput: EditText): Map<EditText, String?> {
        val map: MutableMap<EditText, String?> = HashMap()
        map[taskNameInput] = taskName
        return map
    }

    public override fun onStop() {
        super.onStop()
        presenter.onViewDetached()
    }

    companion object {
        private const val LIST_ID_PARAMETER = "id"
    }
}